src/shader_recompiler/backend/spirv/emit_spirv_context_get_set.cpp

```diff
@@ -219,7 +219,16 @@ void EmitSetAttribute(EmitContext& ctx, IR::Attribute attr, Id value, u32 elemen
         return;
     }
     const Id pointer{OutputAttrPointer(ctx, attr, element)};
-    ctx.OpStore(pointer, ctx.OpBitcast(ctx.F32[1], value));
+    const Id valueF32{ctx.OpBitcast(ctx.F32[1], value)};
+    if (ctx.runtime_info.vs_info.emulate_depth_negative_one_to_one &&
+        attr == IR::Attribute::Position0 && element == 2) {
+        // Adjust the Z value in the shader from [-1, 1] to [0, 1].
+        const Id valueZeroToTwo{ctx.OpFAdd(ctx.F32[1], valueF32, ctx.ConstF32(1.f))};
```

Comment:
I feel this assumes that the written z value is in [-1, 1] range. But games can output any z value together, where z / w is actually what is inside [-1, 1]

Reply:
Hmm, is there any better way to account for this? Since it outputs both `z` and `w` as separate components not sure how I can transform `z / w` as a whole

Reply:
Actually since the clip spaces are `[0, w]` and `[-w, w]` couldn't I just transform `z` with something like:
```
z = (z + w) / 2
```

Reply:
Yeah I feel this transformation will work better

Reply:
You can also implement this as an epilogue to the shader so you have access to both z and w

Reply:
Yep that works, thanks. Had something like that written up real quick but didn't know there was a dedicated epilogue instruction, that's useful.